src/schema/v2/artworksForUser.ts

```diff
@@ -47,33 +52,83 @@ export const getArtistAffinities = async (
   return artistIds
 }
 
+export const getAffinityArtworks = async (
+  artistIds: string[],
+  gravityArgs,
+  context: ResolverContext
+): Promise<any[]> => {
+  if (artistIds.length === 0) return []
+
+  const { size, offset } = gravityArgs
+  const { artworksLoader } = context
+
+  const artworkParams = {
+    artist_ids: artistIds,
+    availability: "for sale",
+    offset,
+    size,
+    sort: "-published_at",
+  }
+
+  const body = await artworksLoader(artworkParams)
+
+  return body
+}
+
+export const getBackfillArtworks = async (
+  remainingSize: number,
+  includeBackfill: boolean,
+  context: ResolverContext
+): Promise<any[]> => {
+  if (!includeBackfill || remainingSize < 1) return []
+
+  const { artworksLoader, setItemsLoader } = context
+
+  // should this be injected in the env? i could also send yet another api call
+  // to get this id via the key of artwork-backfill. could also consider
+  // caching this...
+  const backfillSetId = "SHHH"
+  const response = await setItemsLoader(backfillSetId)
+  const backfillArtworkIds = response.body.map((item) => item._id)
+  const artworkParams = {
+    ids: backfillArtworkIds,
+  }
+  const body = await artworksLoader(artworkParams)
+
+  return body.slice(0, remainingSize)
+}
+
 export const artworksForUser: GraphQLFieldConfig<void, ResolverContext> = {
   description: "A connection of artworks for a user.",
   type: artworkConnection.connectionType,
   args: pageable({
+    // how do i make this not null with a false default?
+    includeBackfill: { type: GraphQLBoolean },
```

Comment:
I can't remember and couldn't easily search for how we are supposed to do this - what I want is a non-null field that defaults to false.

Reply:
You can do `new GraphQLNonNull(GraphQLBoolean)` - but you don't need a default because by explicitly stating that its non-null, one can't send a query without a value, it will error out at the graphql level.

Reply:
Cool updated!